### ISSUE TITLE
[question] building standalone files

### ISSUE THREAD

**Issue by arve0** [arve0/codeclub_lesson_builder#187]
@Jxr227 
> I'm not really sure how this all works. I've got it running, do I need to create index files myself? The main thing I'm after is just the ability to make (standalone) HTML files to support my research as well as PDFs of the same files that can be available as a download.

Yes, you'll need to create an `index.md` file for the "course" you are building. It's metadata is used to create the course name, and scaffold the course index.

This minimum layout should get you going:

```
- codeclub_lesson_builder
- src
  - Turtlebots
    -index.md
    -lesson1.md
```

**Comment by arve0** [comment]
Btw, here's an example: https://github.com/arve0/example_lessons

**Comment by Jxr227** [comment]
Ahhh okay I understand a little better. Sorry to keep asking questions I'm sure you've plenty more to be getting on with, but I certainly appreciate your help!
1) Do I also need an index under src, if I don't then the localhost doesn't point to anything unless I direct it to the Turtlebots subdirectory.
2) The website is in German, is there anyway of modifying that or at least removing the navigation that it uses as a default and only producing .html files of the exercises that I can put links in to from an existing website?
3) Do you have an example index.md that's a little more extensive than the example lesson, so that I can see what sort of things I should be including?

Again, very sorry to keep bothering you, but the help is very much appreciated.

**Comment by arve0** [comment]
> Sorry to keep asking questions I'm sure you've plenty more to be getting on with, but I certainly appreciate your help!

No worries :smile: 

> 1) Do I also need an index under src, if I don't then the localhost doesn't point to anything unless I direct it to the Turtlebots subdirectory.

If your sole purpose is building lessons, you can omit the root `index.md` (I think) and run `gulp build`, which will build the lessons only.

> 2) The website is in German, is there anyway of modifying that or at least removing the navigation that it uses as a default and only producing .html files of the exercises that I can put links in to from an existing website?

You can either [translate the interface](https://github.com/arve0/codeclub_lesson_builder/blob/master/assets/locales), or modify [the layout](https://github.com/arve0/codeclub_lesson_builder/blob/master/layouts/layout.jade).

> 3) Do you have an example index.md that's a little more extensive than the example lesson, so that I can see what sort of things I should be including?

You can have a look at the [Norwegian repo](https://github.com/kodeklubben/oppgaver). It should include the name (`title: name of your corse`), and intro, if you like to use the navigation it supplies.

> 4) If I navigate to the /turtlebot page I can see the index.md, however I can't find lesson1.md even if I've pre-loaded it with a scratch md lesson. Am I doing something wrong?

Push your changes to a repo, and I'll have a quick look.

**Comment by Jxr227** [comment]
I've downloaded the example lessons and got that running, so it's a problem on my side- I'll self diagnose from here so as to not take up your time :P 
Thanks so much for the help. If I produce something effective and use it in my dissertation I'll be sure to credit you for this resource. 

The only issue I've found using your example lessons and running it is cannot get when trying to download PDFs (Which is also present on the IO version of yours) so I'm guessing the functionality doesn't work?
